app/utils/importHelpers.ts

```diff
@@ -325,6 +292,48 @@ async function insertNewAccountsAndCredentials(
     }
 }
 
+/**
+ * Inserts an array of new identities and its attached accounts and credentials. The identities
+ * must be new identities that did not already exist in the database. The accounts and credentials for
+ * each identity in the array will have their identityId reference updated to point to the
+ * correct id for their identity, before they are also inserted into the database.
+ *
+ * Note that for new identities that there cannot be existing account or credentials, so we
+ * can safely insert them without checking if they already exist.
+ * @param newIdentities an array of new identities to be added to the database
+ * @param attachedAccounts all accounts for the current wallet import, without any changes to their identityId
+ * @param attachedCredentials all credentials for the current wallet import, without any changes to their identityId
+ */
+async function insertNewIdentities(
+    newIdentities: Identity[],
+    attachedAccounts: Account[],
+    attachedCredentials: Credential[],
+    addMessage: AddMessage
+) {
+    for (let i = 0; i < newIdentities.length; i += 1) {
+        const { id, ...newIdentity } = newIdentities[i];
+
+        const newIdentityId = (await insertIdentity(newIdentity))[0];
+
+        const {
+            accountsOnIdentity,
+            credentialsOnIdentity,
+        } = findAccountsAndCredentialsOnIdentity(
+            id,
+            attachedAccounts,
+            attachedCredentials
+        );
+
+        // TODO: Find a nice way to avoid the unnecessary check on credentials
```

Comment:
I think the TODO should just be removed as it's okay to have the extra checks on the credentials, it doesn't hurt and it makes for a simpler implementation as we already have it.